homeassistant/components/blink/__init__.py

```diff
@@ -1,157 +1,153 @@
-        )
-    }
+from homeassistant.helpers import config_validation as cv
+
+from .const import (
+    DEFAULT_OFFSET,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    PLATFORMS,
+    SERVICE_REFRESH,
+    SERVICE_SAVE_VIDEO,
+    SERVICE_SEND_PIN,
+    SERVICE_TRIGGER,
 )
 
-SENSOR_SCHEMA = vol.Schema(
-    {
-        vol.Optional(CONF_MONITORED_CONDITIONS, default=list(SENSORS)): vol.All(
-            cv.ensure_list, [vol.In(SENSORS)]
-        )
-    }
-)
+_LOGGER = logging.getLogger(__name__)
 
 SERVICE_TRIGGER_SCHEMA = vol.Schema({vol.Required(CONF_NAME): cv.string})
-
 SERVICE_SAVE_VIDEO_SCHEMA = vol.Schema(
     {vol.Required(CONF_NAME): cv.string, vol.Required(CONF_FILENAME): cv.string}
 )
+SERVICE_SEND_PIN_SCHEMA = vol.Schema({vol.Optional(CONF_PIN): cv.string})
 
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.Schema(
             {
                 vol.Required(CONF_USERNAME): cv.string,
                 vol.Required(CONF_PASSWORD): cv.string,
-                vol.Optional(
-                    CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL
-                ): cv.time_period,
-                vol.Optional(CONF_BINARY_SENSORS, default={}): BINARY_SENSOR_SCHEMA,
-                vol.Optional(CONF_SENSORS, default={}): SENSOR_SCHEMA,
-                vol.Optional(CONF_OFFSET, default=1): int,
-                vol.Optional(CONF_MODE, default=""): cv.string,
+                vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL): int,
             }
         )
     },
     extra=vol.ALLOW_EXTRA,
 )
 
 
-def setup(hass, config):
-    """Set up Blink System."""
-
-    conf = config[BLINK_DATA]
-    username = conf[CONF_USERNAME]
-    password = conf[CONF_PASSWORD]
-    scan_interval = conf[CONF_SCAN_INTERVAL]
-    is_legacy = bool(conf[CONF_MODE] == "legacy")
-    motion_interval = conf[CONF_OFFSET]
-    hass.data[BLINK_DATA] = blinkpy.Blink(
-        username=username,
-        password=password,
-        motion_interval=motion_interval,
-        legacy_subdomain=is_legacy,
-    )
-    hass.data[BLINK_DATA].refresh_rate = scan_interval.total_seconds()
-    hass.data[BLINK_DATA].start()
+async def async_setup(hass, config):
+    """Set up a config entry."""
+    if DOMAIN not in config:
+        return True
 
-    platforms = [
-        ("alarm_control_panel", {}),
-        ("binary_sensor", conf[CONF_BINARY_SENSORS]),
-        ("camera", {}),
-        ("sensor", conf[CONF_SENSORS]),
-    ]
+    conf = config.get(DOMAIN, {})
 
-    for component, schema in platforms:
-        discovery.load_platform(hass, component, DOMAIN, schema, config)
+    if not hass.config_entries.async_entries(DOMAIN):
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN, context={"source": SOURCE_IMPORT}, data=conf
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up Blink via config entry."""
+    if DOMAIN not in hass.data:
```

Comment:
Now you're not saving it when it already exists. Instead, add `hass.data[DOMAIN] = {}` to beginning of `async_setup`. Then you know the dict always exists and you can drop the if-statement.